packages/calypso-e2e/src/lib/components/sidebar-component.ts

```diff
@@ -48,25 +48,43 @@ export class SidebarComponent {
 			await this.openMobileSidebar();
 		}
 
-		try {
-			const itemSelector = `${ selectors.sidebar } :text-is("${ item }"):visible`;
-			await this.scrollItemIntoViewIfNeeded( itemSelector );
+		const itemSelector = `${ selectors.sidebar } :text-is("${ item }"):visible`;
+		await this.scrollItemIntoViewIfNeeded( itemSelector );
 
-			if ( subitem ) {
-				// Click top-level item without waiting for navigation if targeting subitem.
-				await this.page.click( itemSelector );
+		if ( subitem ) {
+			// Click top-level item without waiting for navigation if targeting subitem.
+			await this.page.click( itemSelector );
 
-				const subitemSelector = `:text-is("${ subitem }"):visible:below(${ itemSelector })`;
-				await this.scrollItemIntoViewIfNeeded( subitemSelector );
+			const subitemSelector = `:text-is("${ subitem }"):visible:below(${ itemSelector })`;
+			await this.scrollItemIntoViewIfNeeded( subitemSelector );
 
-				await Promise.all( [ this.page.waitForNavigation(), this.page.click( subitemSelector ) ] );
-			} else {
-				await Promise.all( [ this.page.waitForNavigation(), this.page.click( itemSelector ) ] );
-			}
-			/**
-			 * Retry if the click missed the expected item. This can happen because of
-			 * the lazy-loading nature of the sidenav items.
-			 */
+			await Promise.all( [ this.page.waitForNavigation(), this.page.click( subitemSelector ) ] );
+		} else {
+			await Promise.all( [ this.page.waitForNavigation(), this.page.click( itemSelector ) ] );
+		}
+
+		/**
+		 * Do not attempt to retry if we've navigated outside wp.com as the sidebar
+		 * is no longer present.
+		 */
+		const currentURL = await this.page.url();
+		if ( ! currentURL.startsWith( 'https://wordpress.com' ) ) {
```

Comment:
I think this is a good trade-off, but we should be aware that it'll eventually prevent us from running CI tests against a different URL, not a huge deal now as we haven't really done that yet, but there is a parameter for that in the WPCom Tests->Gutenberg tests buildType, called Test URL. This could be a problem if we decide to migrate this param over to the Playwright buildTypes. cc @scinos

Reply:
Yeah, that value would need to be env-dependant. I'm wondering if we still need the retry logic though (after the latest sidebar tweaks), so maybe we can revisit that and in case it's not, this check will be gone as well.